File: GitInsights/Domain/Entities/Repository.cs
using System;
using System.Collections.Generic;

namespace Domain.Entities
{
    public class Repository
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string HtmlUrl { get; set; }     
        public string Description { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public int StargazersCount { get; set; }
        public int LanguageId { get; set; }
        public Language Language { get; set; }

        public Repository()
        {
            
        }
        public Repository(string name, string htmlUrl, string description, DateTime createdAt, DateTime updatedAt, int stargazersCount, Language language)
        {
            ValidateParams(name, htmlUrl, description, createdAt, updatedAt, stargazersCount, language);
            Name = name;
            HtmlUrl = htmlUrl;
            Description = description;
            CreatedAt = createdAt;
            UpdatedAt = updatedAt;
            StargazersCount = stargazersCount;
            Language = language;
            LanguageId = language.Id;
        }

        private void ValidateParams(string name, string htmlUrl, string description, DateTime createdAt, DateTime updatedAt, int stargazersCount, Language language)
        {
            if (string.IsNullOrWhiteSpace(name))
            {
                throw new ArgumentException("Invalid parameter!", "Name");
            }

            if (string.IsNullOrWhiteSpace(htmlUrl))
            {
                throw new ArgumentException("Invalid parameter!", "HtmlUrl");
            }

            if (string.IsNullOrWhiteSpace(description))
            {
                throw new ArgumentException("Invalid parameter!", "Description");
            }

            if (createdAt == null || createdAt > DateTime.Now)
            {
                throw new ArgumentException("Invalid parameter!", "CreatedAt");
            }

            if (updatedAt == null || updatedAt > DateTime.Now)
            {
                throw new ArgumentException("Invalid parameter!", "UpdatedAt");
            }

            if (stargazersCount <= 0)
            {
                throw new ArgumentException("Invalid parameter!", "StargazersCount");
            }

            if (language == null)
            {
                throw new ArgumentException("Invalid parameter!", "Language");
            }
        }
    }
}

File: back-end/src/main/java/com/github/integration/backend/persistence/entities/Repository.java
package com.github.integration.backend.persistence.entities;

import lombok.*;

import javax.persistence.*;
import java.util.Date;

@NoArgsConstructor
@AllArgsConstructor
@ToString
@Builder
@Getter
@Setter
@Entity
public class Repository {

    @Id
    private Long id;

    private String nodeId;

    private String name;

    private String fullName;

    private Boolean privateRepository;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private Owner owner;

    private String htmlUrl;

    @Column(columnDefinition="text")
    private String description;

    private Boolean fork;

    private String url;

    private String forksUrl;

    private String keysUrl;

    private String collaboratorsUrl;

    private String teamsUrl;

    private String hooksUrl;

    private String issueEventsUrl;

    private String eventsUrl;

    private String assigneesUrl;

    private String branchesUrl;

    private String tagsUrl;

    private String blobsUrl;

    private String gitTagsUrl;

    private String gitRefsUrl;

    private String treesUrl;

    private String statusesUrl;

    private String languagesUrl;

    private String stargazersUrl;

    private String contributorsUrl;

    private String subscribersUrl;

    private String subscriptionUrl;

    private String commitsUrl;

    private String gitCommitsUrl;

    private String commentsUrl;

    private String issueCommentUrl;

    private String contentsUrl;

    private String compareUrl;

    private String mergesUrl;

    private String archiveUrl;

    private String downloadsUrl;

    private String issuesUrl;

    private String pullsUrl;

    private String milestonesUrl;

    private String notificationsUrl;

    private String labelsUrl;

    private String releasesUrl;

    private String deploymentsUrl;

    private Date createdAt;

    private Date updatedAt;

    private Date pushedAt;

    private String gitUrl;

    private String sshUrl;

    private String cloneUrl;

    private String svnUrl;

    private String homepage;

    private Long size;

    private Integer stargazersCount;

    private Integer watchersCount;

    private String language;

    private Boolean hasIssues;

    private Boolean hasProjects;

    private Boolean hasDownloads;

    private Boolean hasWiki;

    private Boolean hasPages;

    private Integer forksCount;

    private String mirrorUrl;

    private Boolean archived;

    private Boolean disabled;

    private Integer openIssuesCount;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private License license;

    private Integer forks;

    private Integer openIssues;

    private Integer watchers;

    private String defaultBranch;

    private Double score;

}